lib/valueflow.cpp

```diff
@@ -2318,6 +2318,8 @@ struct OppositeExpressionAnalyzer : ExpressionAnalyzer {
     {}
 
     virtual bool match(const Token* tok) const OVERRIDE {
+        if (tok->isName() && !tok->isBoolean())
```

Comment:
It seems this logic would be better in `isOppositeCond` function, instead of just here in valueflow.

Reply:
Indeed. I ended up writing a more complex logic, because what isOppositeCond needs to know is if the tokens that are passed to it really are conditions (~used as boolean).
I hope I did not forgot some case, feel free to give me more test cases to add. ;-)